src/App.tsx

```diff
@@ -1,27 +1,49 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
+import { Good } from './types/Good';
+import { getAll, get5First, getRedGoods } from './api/goods';
 // or
 // import * as goodsAPI from './api/goods';
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+  async function handlerLoadGood(f: () => Promise<Good[]>) {
+    const result = await f();
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+    setGoods(result);
+  }
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+  return (
+    <div className="App">
+      <h1>Dynamic list of Goods</h1>
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+      <button
+        type="button"
+        data-cy="all-button"
+        onClick={() => handlerLoadGood(getAll)}
+      >
+        Load all goods
+      </button>
 
-    <GoodsList goods={[]} />
-  </div>
-);
+      <button
+        type="button"
+        data-cy="first-five-button"
+        onClick={() => handlerLoadGood(get5First)}
+      >
+        Load 5 first goods
+      </button>
+
+      <button
+        type="button"
+        data-cy="red-button"
+        onClick={() => handlerLoadGood(getRedGoods)}
+      >
+        Load red goods
+      </button>
+
+      <GoodsList goods={goods || []} />
```

Comment:
```suggestion
      <GoodsList goods={goods} />
```
redundant
goods has initial state empty array